### PR TITLE
feat: adding immediate nullification pattern

### PR DESCRIPTION
**Pull request by 0xng**
# 🤖 Linear
Closes AZT-91, AZT-92

### CODE REVIEW

File: patterns/immediate-nullification/README.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            id: _id,
+            other_member: _other_member,
+            other_member_2: _other_member_2,
+            header: NoteHeader::empty(),
+        }
+    }
+
+	/// @notice Nullifying hash generator
+    /// @param _id_ The id to nullify
+    pub fn generate_round_randomness_nullifier(_id: Field) -> Field {
```
Comment (simon-something) [inline patterns/immediate-nullification/README.md]: we keep "round" in the name?

File: patterns/immediate-nullification/src/test/e2e.test.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        .send()
+        .wait();
+  
+      immediateNullificationContract = deploymentReceipt.contract;
+    }, 200_000);
+
+    describe("create_note(...)", () => {
+        let exampleNotes: ExtendedNote[];
+
+        it("should mine the transaction", async () => {
```
Comment (simon-something) [inline patterns/immediate-nullification/src/test/e2e.test.ts]: "transaction should not revert" or something similar? iirc, they left a comment once about it - what we care is not reverting (i know, I nit now)

File: patterns/immediate-nullification/src/types/example_note.nr [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+            owner,
+            randomness,
+            header: NoteHeader::empty(),
+        }
+    }
+
+    /// @notice Parameter-nullifying hash generator
+    /// @param _owner The owner of the note to nullify
+    /// @param _randomness The random parameter provided by the note's creator
+    pub fn generate_round_randomness_nullifier(_owner: AztecAddress, _randomness: Field) -> Field {
```
Comment (simon-something) [inline patterns/immediate-nullification/src/types/example_note.nr:60]: same, round?

File: patterns/note-sharing/src/types/shared_note.nr
Comment (simon-something) [inline patterns/note-sharing/src/types/shared_note.nr:1]: 👻

**Review by simon-something — DISMISSED** [review]

**Review by simon-something — APPROVED** [review]

**Review by skeletor-spaceman — APPROVED** [review]

**Pull request merged**